Added ability to show item drop tables for detail object display

savage13 opened the pull request:
If the `drop_table` value exists in the returned object from the radar server, it formats the drop table in a `pre` tag with some minor formatting.  Format should be similar to the original raw text.

Nothing is currently done with `drop_tables`.  It would allow a bit of customization if the Actor can be `iced` or `burned` or `happyed`, ...

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zeldamods/objmap/29)
<!-- Reviewable:end -->

leoetlino commented:
Hmm, I'm not sure about the design -- I'm a bit wary of making the object structure bigger and of adding too much logic in the object server. Couldn't we instead make the client fetch the correct DropTable (which could be a simple JSON/YAML: [example](https://github.com/leoetlino/botw/blob/master/Actor/DropTable/TBox_Field_Sandworm.drop.yml)) based on the DropTable/ArrowName/whatever properties?

Also, would it be possible to have the generator script for the drop table data in the repo? And does it account for the Actor -> DropTable mapping (which is in ActorLink) properly?

One last thing: we don't have a linter to enforce code style but please try to ensure newly added code follows the existing code style and naming conventions.

---

Relevant PR in botwradar: https://github.com/zeldamods/radar/pull/2

savage13 commented:
Yes on all three points. 

Will create a new route for the drop table searching by the unit_config_name and specialized table names.

Will see what I can do with dynamically requesting the drop data in the detail view.

Sorry about the code style / naming conventions  :)

savage13 commented:
For `radar`:
I modified the build.ts script to read in all ActorLink/*.yml files and then loads the specified DropTables. These are all loaded into the `drop_table` db table.  The json file is now gone.  I only added a few example files for the ActorLink and DropTables.  

I checked all of the UnitConfigNames (from the `objs` table) and they all have associated DropTable files.  There are a number of DropTables without associated UnitConfigNames (like Guardian_A_Leg) in the `objs` table because they do not exist on the Map explicitly.

New routes added as `/drop/unit_config_name` and `/drop/unit_config_name/table_name`. These should only be touched when view the object detail (similar to genGroups).

For `objmap`:
Modified based on your suggestions.  Thanks, the code is cleaner.

Hopefully I am following the code and naming guidelines.  
Let me know if I missed something.

leoetlino commented:
This PR looks mostly fine to me. I'll review this more carefully after the backend PR is merged :)

leoetlino reviewed (changes requested):
> Reviewed 2 of 3 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/zeldamods/objmap/29)* status: all files reviewed, 3 unresolved discussions (waiting on @savage13)

___
*[src/components/AppMapDetailsObj.ts, line 359 at r3](https://reviewable.io/reviews/zeldamods/objmap/29#-Mf39fdw-nvzB3Ti2MRZ:-Mf39fdw-nvzB3Ti2MR_:bm2mr1f) ([raw file](https://github.com/zeldamods/objmap/blob/3ee7c8be5fb91358e3d8997c718e0437aa2c6846/src/components/AppMapDetailsObj.ts#L359)):*
> ```XML
>   }
> 
>   getName(name: string) {
> ```

This can be a static method (since it doesn't use `this` at all). 
___
*[src/components/AppMapDetailsObj.ts, line 367 at r3](https://reviewable.io/reviews/zeldamods/objmap/29#-Mf39qW91Dlb5qht0M1A:-Mf39qW91Dlb5qht0M1B:b3jfi2r) ([raw file](https://github.com/zeldamods/objmap/blob/3ee7c8be5fb91358e3d8997c718e0437aa2c6846/src/components/AppMapDetailsObj.ts#L367)):*
> ```XML
>   }
> 
>   dropTableFormat() : string {
> ```

formatDropTable
___
*[src/components/AppMapDetailsObj.ts, line 384 at r3](https://reviewable.io/reviews/zeldamods/objmap/29#-Mf3B03N7eW-Ie_KQ-fh:-Mf3B03N7eW-Ie_KQ-fi:b-9fikyp) ([raw file](https://github.com/zeldamods/objmap/blob/3ee7c8be5fb91358e3d8997c718e0437aa2c6846/src/components/AppMapDetailsObj.ts#L384)):*
> ```XML
>     }
>     return lines.join("\n");
>   }
> ```

new line after this line please


<!-- Sent from Reviewable.io -->

savage13 commented:
Formatting / Function name fixes applied

leoetlino reviewed (approved):
> Reviewed 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/zeldamods/objmap/29)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @savage13)



<!-- Sent from Reviewable.io -->
